### DISCUSSION TITLE
Filtering & Sorting by Model append Attributes

### DISCUSSION THREAD

**Discussion by shingsoso** [spatie/laravel-query-builder#459] [Q&A] [answered]
Is there any way to implement filtering & sorting by the model appended attributes ? 
For example : 

```
protected $appends = [
        'average_rating', 'total_review_count', 'thumbnail', 'total_visits_count'
    ];
public function getTotalReviewCountAttribute()
{
        return $this->reviews()->count();
}
```
I want to do sorting on appended attributes such as average_rating , total_review_count, etc

**Comment by ashgibson** [comment]
Appended attributes are added to the model after the database is queried. You will need to do a sort on the collection returned from the query.

```
$collection = QueryBuilder::for(MyModel::class)->get();
$collection->sortBy('average_rating');
```

**Comment by AlexVanderbist** [comment, marked as answer]
Hi! What @ashgibson says is true. The query builder builds Eloquent queries, it doesn't modify collections. Adding support for appended models might've been a mistake from the beginning 😅 

I'd actually recommend looking into SQL and Eloquent queries more. Most of the extra attributes I add to models can actually be written as selects/joins/relations/etc. 

For your total review count example we actually support selecting and sorting by relationship count:
```php
$users = QueryBuilder::for(User::class)
    ->allowedIncludes([
        'posts', // allows including `posts` or `postsCount`
    ])
    ->allowedSorts('posts_count'); 
```

Good luck!
